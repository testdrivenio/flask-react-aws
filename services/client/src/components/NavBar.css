.link:hover {
  cursor: pointer;
  background-color: #292929;
  color: #f5f5f5;
}
