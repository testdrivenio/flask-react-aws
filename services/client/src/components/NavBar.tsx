import React from "react";
import {
  Box,
  Flex,
  Heading,
  Link,
  Spacer,
  IconButton,
  Stack,
  Collapse,
  useDisclosure,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import { Link as RouterLink } from "react-router-dom";

interface NavBarProps {
  title: string;
  logoutUser: () => void;
  isAuthenticated: () => boolean;
}

const NavBar: React.FC<NavBarProps> = ({
  title,
  logoutUser,
  isAuthenticated,
}) => {
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Box bg="gray.800" px={4}>
      <Flex h={16} alignItems="center" justifyContent="space-between">
        <Flex alignItems="center">
          <Heading as="h1" size="lg" className="navbar-item nav-title" mr={4}>
            <Link
              as={RouterLink}
              to="/"
              _hover={{ textDecoration: "none" }}
              color="white"
            >
              {title}
            </Link>
          </Heading>
        </Flex>
        <Spacer />
        <Flex alignItems="center" display={{ base: "none", md: "flex" }}>
          <NavLinks logoutUser={logoutUser} isAuthenticated={isAuthenticated} />
        </Flex>
        <IconButton
          display={{ base: "flex", md: "none" }}
          onClick={onToggle}
          icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
          variant="outline"
          aria-label="Toggle Navigation"
          color="white"
        />
      </Flex>

      <Collapse in={isOpen} animateOpacity>
        <Stack mt={2} display={{ md: "none" }} bg="gray.800" p={4} spacing={4}>
          <NavLinks logoutUser={logoutUser} isAuthenticated={isAuthenticated} />
        </Stack>
      </Collapse>
    </Box>
  );
};

const NavLinks: React.FC<{
  logoutUser: () => void;
  isAuthenticated: () => boolean;
}> = ({ logoutUser, isAuthenticated }) => (
  <>
    <Link as={RouterLink} to="/about" mr={4} color="white">
      About
    </Link>
    {isAuthenticated() ? (
      <>
        <Link as={RouterLink} to="/status" mr={4} color="white">
          User Status
        </Link>
        <Link
          as="span"
          onClick={logoutUser}
          mr={4}
          color="white"
          cursor="pointer"
        >
          Log Out
        </Link>
      </>
    ) : (
      <>
        <Link as={RouterLink} to="/register" mr={4} color="white">
          Register
        </Link>
        <Link as={RouterLink} to="/login" mr={4} color="white">
          Log In
        </Link>
      </>
    )}
  </>
);

export default NavBar;
