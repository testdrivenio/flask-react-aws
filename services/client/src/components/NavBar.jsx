import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./NavBar.css";

const titleStyle = {
  fontWeight: "bold"
};

const NavBar = props => {
  let menu = (
    <div className="navbar-menu">
      <div className="navbar-start">
        <Link to="/about" className="navbar-item" data-testid="nav-about">
          About
        </Link>
      </div>
      <div className="navbar-end">
        <Link to="/register" className="navbar-item" data-testid="nav-register">
          Register
        </Link>
        <Link to="/login" className="navbar-item" data-testid="nav-login">
          Log In
        </Link>
      </div>
    </div>
  );
  if (props.isAuthenticated()) {
    menu = (
      <div className="navbar-menu">
        <div className="navbar-start">
          <Link to="/about" className="navbar-item" data-testid="nav-about">
            About
          </Link>
          <Link to="/status" className="navbar-item" data-testid="nav-status">
            User Status
          </Link>
        </div>
        <div className="navbar-end">
          <span
            // eslint-disable-next-line react/jsx-handler-names
            onClick={props.logoutUser}
            className="navbar-item link"
            data-testid="nav-logout"
          >
            Log Out
          </span>
        </div>
      </div>
    );
  }
  return (
    <nav
      className="navbar is-dark"
      role="navigation"
      aria-label="main navigation"
    >
      <section className="container">
        <div className="navbar-brand">
          <Link to="/" className="navbar-item nav-title" style={titleStyle}>
            {props.title}
          </Link>
          <span
            className="nav-toggle navbar-burger"
            onClick={() => {
              let toggle = document.querySelector(".nav-toggle");
              let menu = document.querySelector(".navbar-menu");
              toggle.classList.toggle("is-active");
              menu.classList.toggle("is-active");
            }}
          >
            <span />
            <span />
            <span />
          </span>
        </div>
        {menu}
      </section>
    </nav>
  );
};

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
  logoutUser: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.func.isRequired
};

export default NavBar;
