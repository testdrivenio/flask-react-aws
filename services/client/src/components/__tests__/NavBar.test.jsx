import React from 'react';
import { cleanup, wait } from '@testing-library/react';

import NavBar from '../NavBar';

afterEach(cleanup);

describe('when unauthenticated', () => {
  const props = {
    title: 'Hello, World!',
    logoutUser: () => { return true },
    isAuthenticated: jest.fn().mockImplementation(() => false),
  }

  it('renders the default props', async() => {
    const { getByText, findByTestId } = renderWithRouter(<NavBar {...props} />);
    expect(getByText(props.title)).toHaveClass('nav-title');
    await wait(() => {
      expect(props.isAuthenticated).toHaveBeenCalledTimes(1);
    });
    expect((await findByTestId('nav-about')).innerHTML).toBe('About');
    expect((await findByTestId('nav-register')).innerHTML).toBe('Register');
    expect((await findByTestId('nav-login')).innerHTML).toBe('Log In');
  });

  it("renders", () => {
    const { asFragment } = renderWithRouter(<NavBar {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('when authenticated', () => {
  const props = {
    title: 'Hello, World!',
    logoutUser: () => { return true },
    isAuthenticated: jest.fn().mockImplementation(() => true),
  }

  it('renders the default props', async() => {
    const { getByText, findByTestId } = renderWithRouter(<NavBar {...props} />);
    expect(getByText(props.title)).toHaveClass('nav-title');
    await wait(() => {
      expect(props.isAuthenticated).toHaveBeenCalledTimes(1);
    });
    expect((await findByTestId('nav-about')).innerHTML).toBe('About');
    expect((await findByTestId('nav-status')).innerHTML).toBe('User Status');
    expect((await findByTestId('nav-logout')).innerHTML).toBe('Log Out');
  });

  it("renders", () => {
    const { asFragment } = renderWithRouter(<NavBar {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
