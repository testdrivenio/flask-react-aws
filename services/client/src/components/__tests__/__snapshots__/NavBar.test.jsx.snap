// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when authenticated renders 1`] = `
<DocumentFragment>
  <nav
    aria-label="main navigation"
    class="navbar is-dark"
    role="navigation"
  >
    <section
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item nav-title"
          href="/"
          style="font-weight: bold;"
        >
          Hello, World!
        </a>
        <span
          class="nav-toggle navbar-burger"
        >
          <span />
          <span />
          <span />
        </span>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            data-testid="nav-about"
            href="/about"
          >
            About
          </a>
          <a
            class="navbar-item"
            data-testid="nav-status"
            href="/status"
          >
            User Status
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <span
            class="navbar-item link"
            data-testid="nav-logout"
          >
            Log Out
          </span>
        </div>
      </div>
    </section>
  </nav>
</DocumentFragment>
`;

exports[`when unauthenticated renders 1`] = `
<DocumentFragment>
  <nav
    aria-label="main navigation"
    class="navbar is-dark"
    role="navigation"
  >
    <section
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item nav-title"
          href="/"
          style="font-weight: bold;"
        >
          Hello, World!
        </a>
        <span
          class="nav-toggle navbar-burger"
        >
          <span />
          <span />
          <span />
        </span>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            data-testid="nav-about"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <a
            class="navbar-item"
            data-testid="nav-register"
            href="/register"
          >
            Register
          </a>
          <a
            class="navbar-item"
            data-testid="nav-login"
            href="/login"
          >
            Log In
          </a>
        </div>
      </div>
    </section>
  </nav>
</DocumentFragment>
`;
